Python/ceval_gil.h

```diff
@@ -206,6 +246,20 @@ take_gil(struct _ceval_runtime_state *ceval, PyThreadState *tstate)
 
         unsigned long interval = (gil->interval >= 1 ? gil->interval : 1);
         COND_TIMED_WAIT(gil->cond, gil->mutex, interval, timed_out);
+
+        if (thread_must_exit(tstate)) {
+            /* bpo-36475: If Py_Finalize() has been called and tstate is not
+               the thread which called Py_Finalize(), exit immediately the
+               thread.
+
+               This code path can be reached by a daemon thread which continues
+               to run after wait_for_thread_shutdown() and before Py_Finalize()
+               completes. For example, when _PyImport_Cleanup() executes Python
+               code. */
+            MUTEX_UNLOCK(gil->mutex);
```

Comment:
I'm not sure if anyone else should be done here.

Reply:
For example, is COND_SIGNAL(gil->cond) needed to wake up the next thread waiting on COND_TIMED_WAIT()?